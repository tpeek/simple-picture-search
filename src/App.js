import React, { useState, useEffect } from 'react';
import './App.css';
import Search from './Search';
import SearchResult from './SearchResult';

const API_KEY = process.env.REACT_APP_API_KEY;

function App() {
  const [results, setResults] = useState([]);

  const fetchSearchResults = (keyword = '', category = '') => {
    fetch(
      `https://pixabay.com/api/?key=${API_KEY}&q=${keyword}&category=${category}&image_type=photo&per_page=10`
    ).then((result) => {
      result.json().then((json) => {
        console.log(json);
        setResults(json.hits);
      });
    });
  };

  useEffect(() => {
    fetchSearchResults();
  }, []);

  return (
    <div className="App">
      <Search onSearch={fetchSearchResults} />
      <ul>
        {results.map(({ id, ...rest }) => (
          <SearchResult key={id} id={id} {...rest} />
        ))}
      </ul>
    </div>
  );
}

export default App;
