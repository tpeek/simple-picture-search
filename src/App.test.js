import React from 'react';
import App from './App';

const setup = () => shallow(<App />);

describe('<App />', () => {
  it('renders snapshot correctly', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
