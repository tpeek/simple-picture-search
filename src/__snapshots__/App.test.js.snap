// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders snapshot correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="makeStyles-leftPanel-2"
  >
    <Search
      onSearch={[Function]}
    />
    <SearchResults
      results={Array []}
      saveResult={[Function]}
      savedResults={Array []}
    />
  </div>
  <div
    className="makeStyles-rightPanel-3"
  >
    <SavedArea
      savedResults={Array []}
    />
  </div>
</div>
`;
